Backpack/CardButton/Classes/SaveCardButton.swift

```diff
@@ -0,0 +1,254 @@
+                name: icons.defaultIcon,
+                color: normalIconColor(style: style, checked: checked)
+            )
+            setImage(largeIconNormal, for: .normal)
+
+            let largeIconHighlighted = BPKIcon.makeLargeIcon(
+                name: icons.defaultIconHighlighted,
+                color: highlightedIconColor(style: style, checked: checked)
+            )
+            setImage(largeIconHighlighted, for: .highlighted)
+        case .small:
+            let smallIconNormal = BPKIcon.makeSmallIcon(
+                name: icons.smallIcon,
+                color: normalIconColor(style: style, checked: checked)
+            )
+            setImage(smallIconNormal, for: .normal)
+
+            let smallIconHighlighted = BPKIcon.makeSmallIcon(
+                name: icons.smallIconHighlighted,
+                color: highlightedIconColor(style: style, checked: checked)
+            )
+            setImage(smallIconHighlighted, for: .highlighted)
+        }
+    }
+
+    private func normalIconColor(style: BPKCardButtonStyle, checked: Bool) -> UIColor {
+        let colors = checked ? Self.checkedColor : Self.uncheckedColor
+
+        switch style {
+        case .onDark:
+            return colors.iconColorOnDark
+        case .contained:
+            return colors.iconColorContained
+        case .default:
+            return colors.iconColorDefault
+        }
+    }
+
+    private func highlightedIconColor(style: BPKCardButtonStyle, checked: Bool) -> UIColor {
+        let colors = checked ? Self.checkedColor : Self.uncheckedColor
+
+        switch style {
+        case .onDark:
+            return colors.highlightedIconColorOnDark
+        case .contained:
+            return colors.highlightedIconColorContained
+        case .default:
+            return colors.highlightedIconColorDefault
+        }
+    }
+
+    private func updateBackgroundCircleView() {
+        if viewConfigurator.shouldAddButtonBackground(style: style) {
+            if containedBackgroundCircle.superview == nil {
+                addSubview(containedBackgroundCircle)
+                let circleSize = viewConfigurator.buttonBackgroundSize(size: size)
+                containedBackgroundCircle.layer.cornerRadius = circleSize / 2
+
+                NSLayoutConstraint.activate([
+                    containedBackgroundCircle.widthAnchor.constraint(equalToConstant: circleSize),
+                    containedBackgroundCircle.heightAnchor.constraint(equalToConstant: circleSize),
+                    containedBackgroundCircle.centerXAnchor.constraint(equalTo: centerXAnchor),
+                    containedBackgroundCircle.centerYAnchor.constraint(equalTo: centerYAnchor)
+                ])
+                imageView.map { bringSubviewToFront($0) }
+            }
+        } else {
+            if containedBackgroundCircle.superview != nil {
+                containedBackgroundCircle.removeFromSuperview()
+            }
+        }
```

Comment:
Could we flip this if, by using 

```
guard viewConfigurator.shouldAddButtonBackground(style: style) else { 
    if containedBackgroundCircle.superview != nil {
           containedBackgroundCircle.removeFromSuperview()
     }
}
```

Reply:
sure

Reply:
The func was refactored. cc00541f816323d9ee970829f3462d4ec23a4903